### ISSUE TITLE
Determining if a token is the first on a line

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#314]
During my ongoing work on https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/261, I noticed a potential problem with [CommonFormattingHelpers.IsFirstTokenOnLine](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/CommonFormattingHelpers.cs,fcf0d57b28a05a1e).

Consider the following code:

```csharp
int x = Foo(a
, b);
```

In this case, `IsFirstTokenOnLine` returns true for the `,` token. However, if the input is modified to the following:

```csharp
int x = Foo(a // a comment
, b);
```

The code returns false for the `,` token. This occurs because non-whitespace characters appear between the `,` token and the preceding token `a`.

**Comment by heejaechang** [comment]
I guess there is no end of line trivia there....

**Comment by mattwar** [comment]
The logic in  [CommonFormattingHelpers.IsFirstTokenOnLine](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/CommonFormattingHelpers.cs,fcf0d57b28a05a1e) is flawed.  It makes some nonsense test using string.IsNullOrWhiteSpace for the text between the two tokens, but this doesn't consider that the previous token (on the previous) line has trailing trivia (on that same line).

On a side note, this method is rather expensive, using GetText() and getting lines from positions, which entails a bunch of work.  It would be better if it did actually search in the trivia between the tokens for a line break.

**Comment by heejaechang** [comment]
yep, it looks like no one uses it. I probably wrote it long time ago for indenter where SourceText should be in memory. I will remove the code.

**Comment by heejaechang** [comment]
I removed this code (CommonFormattingHelpers.IsFirstTokenOnLine<http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/CommonFormattingHelpers.cs,fcf0d57b28a05a1e>) since nobody in Roslyn is using.

And yes, the code should have walked backward from tree.GetText().GetText(previousToken, token) until it either see non-whitespace char or new line. Anyway, the code is now removed.


-          heejae
